src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,320 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * BMP388 Driver author: Dominic Clifton
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "build/build_config.h"
+#include "build/debug.h"
+
+#include "common/maths.h"
+
+#include "barometer.h"
+
+#include "drivers/bus.h"
+#include "drivers/bus_i2c.h"
+#include "drivers/bus_i2c_busdev.h"
+#include "drivers/bus_spi.h"
+#include "drivers/io.h"
+#include "drivers/time.h"
+#include "drivers/nvic.h"
+
+#include "barometer_bmp388.h"
+
+//#define DEBUG_BARO_BMP388
+
+#if defined(USE_BARO) && (defined(USE_BARO_BMP388) || defined(USE_BARO_SPI_BMP388))
+
+// see Datasheet 3.11.1 Memory Map Trimming Coefficients
+typedef struct bmp388_calib_param_s {
+    uint16_t T1;
+    uint16_t T2;
+    int8_t T3;
+    int16_t P1;
+    int16_t P2;
+    int8_t P3;
+    int8_t P4;
+    uint16_t P5;
+    uint16_t P6;
+    int8_t P7;
+    int8_t P8;
+    int16_t P9;
+    int8_t P10;
+    int8_t P11;
+} __attribute__((packed)) bmp388_calib_param_t;
+
+static uint8_t bmp388_chip_id = 0;
+STATIC_UNIT_TESTED bmp388_calib_param_t bmp388_cal;
+// uncompensated pressure and temperature
+uint32_t bmp388_up = 0;
+uint32_t bmp388_ut = 0;
+STATIC_UNIT_TESTED int64_t t_lin = 0;
+
+static void bmp388_start_ut(baroDev_t *baro);
+static void bmp388_get_ut(baroDev_t *baro);
+static void bmp388_start_up(baroDev_t *baro);
+static void bmp388_get_up(baroDev_t *baro);
+
+STATIC_UNIT_TESTED void bmp388_calculate(int32_t *pressure, int32_t *temperature);
```

Comment:
Did you forget to add the unit test for this? :wink:

Reply:
Yes!  I forgot to `git add` it, thanks!

Reply:
done!